skimage/filters/_frangi.py

```diff
@@ -0,0 +1,190 @@
+import numpy as np
+
+__all__ = ['frangi', 'hessian']
+
+
+def _frangi_hessian_common_filter(image, scale, scale_step, beta1, beta2):
+    """This is an intermediate function for Frangi and Hessian filters.
+
+    Shares the common code for Frangi and Hessian functions.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Array with input image data.
+    scale : tuple of floats, optional
+        The range of sigmas used.
+    scale_step : float, optional
+        Step size between sigmas.
+    beta1 : float, optional
+        Frangi correction constant.
+    beta2 : float, optional
+        Frangi correction constant.
+    black_ridges : boolean, optional
+        If True (default), detects black ridges, if False - white ones.
+
+    Returns
+    -------
+    filtered_list : list
+        List of pre-filtered images.
+
+    """
+
+    # Import has to be here due to circular import error
+    from ..feature import hessian_matrix, hessian_matrix_eigvals
+
+
+    sigmas = np.arange(scale[0], scale[1], scale_step)
+
+    if np.any(np.asarray(sigmas) < 0.0):
+        raise ValueError("Sigma values less than zero are not valid")
+
+    beta1 = 2 * beta1 ** 2
+    beta2 = 2 * beta2 ** 2
+
+    filtered_array = np.zeros((len(sigmas), np.shape(image)[0],
```

Comment:
This is just `filtered_array = np.zeros((len(sigmas), *image.shape))`.
Same for `lambdas_array`.